src/mfs.jl

```diff
@@ -0,0 +1,239 @@
+[`MFSBruteForce`](@ref) is an algorithm based on randomised search with two consequent steps: random initialization and sphere radius reduction.
+[`MFSBlackBoxOptim`](@ref) is an algorithm based derivative free optimization. It uses BlackBoxOptim package to find the best shock.
+
+`mapper::AttractorMapper` one of available in Attractors.jl [`AttractorMapper`](@ref) constructed with respect to dynamical system.
+`u0` vector containing coordinates of initial point to be tested.
+`search_area` array of tuples or a tuple, by specifying single value e.g.  `(-1.5, 1.5)` or `[(-1.5, 1.5)]` you specify this value for each of the variables, or you may specify separetely 
+search area for each of your states by, for example in 2 dimensions, `[(-1.5, 1.5), (-1, 1)]`.
+`algorithm` one of the two algorithms available: [`MFSBruteForce`](@ref) or [`MFSBlackBoxOptim`](@ref).
+
+## Setup 
+While working with high dimensional systems, we recommend using [`MFSBlackBoxOptim`](@ref) algorithm, as it will give more precise results. While you can use [`MFSBruteForce`](@ref) algorithm 
+to quickly obtain some guess which you can pass as the argument in initialization of [`MFSBlackBoxOptim`](@ref) algorithm to optimize it and obtain exact results more efficiently. Or you can 
+also pass [`MFSBruteForce`](@ref) algorithm as the argument in initialization of [`MFSBlackBoxOptim`](@ref) with parameter random_algo.
+We recommend to pay attention to setup parameters of the algorithms, as default settings may not be sufficient to obtain precise results in some cases. Parameters `MaxSteps` and `search_area` are crucial in initialization
+of [`MFSBlackBoxOptim`](@ref) algorithm. The higher the `MaxSteps` value, the more precise results may be obtained with the cost of longer computation time. By manually decresing approximaton of `search_area`, 
+you may significantly optimize algorithm's performance.
+
+## Description
+The minimal fatal shock algorithm is used to find the smallest perturbation of the initial point `u0` that will 
+lead to a different basin of attraction. It is inspired by the paper "Minimal fatal shocks in multistable complex networks"[^MFS2020] by L. Halekotte and U. Feudel.
+
+The randomized algorithm consists of two steps: random initialization and sphere radius reduction. On the first step, the algorithm generates random pertubations
+of the initial point `u0` and with the best obtained pertubation it proceeds to the second step. On the second step, the algorithm generates random pertubations 
+on the surface of the hypersphere with radius equal to the norm of the best pertubation found in the first step. It reduces the radius of the hypersphere to continue
+searching for the better result within smaller radius.  
+
+The optimization algorithm uses BlackBoxOptim.jl package to find the best shock. It is based on derivative free optimization and uses the objective function with penalties
+to find the minimal fatal shock.
```

Comment:
You should port these into their respective documentation strings for each algorithm.